### ISSUE TITLE
No more free travis

### ISSUE THREAD

**Issue by kanongil** [hapijs/hapi#4180]
I don't think there is a another place to post org-wide issues, so I will post here.

As [mentioned in hapi hour](https://hapihour.slack.com/archives/C6BMHNBLM/p1604327144134800?thread_ts=1604002517.127800&cid=C6BMHNBLM), Travis is [deprecating](https://mailchi.mp/3d439eeb1098/travis-ciorg-is-moving-to-travis-cicom) their free for public repos model. For .com customers, this [has already been done](https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing), while .org (which hapi uses) is gone by the end of the year.

As I see it, there are two paths forward

1. Switch to the .com travis, and periodically apply for open-source credits / get a sponsor.
1. Switch to github actions, which is still free for public repos.

The first option requires less initial work, as the `.travis.yml` configs can be re-used without modification. However, it will periodically require work to secure more credits (which might happen at an inconvenient time).

The second option will require new GH actions configuration, and a per-repo setup of this. This might be simple using [workflow templates](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/sharing-workflows-with-your-organization). Once setup, it should not require any significant maintenance (beyond what maintaining a `.travis.yml` file does).

**Comment by dominykas** [comment]
If I read that correctly, Travis will still give 1000 free minutes? (GH gives you 2500, but workflows are a lot less maintainable)

**Comment by dominykas** [comment]
Or is that a one-off 1000 minutes, not monthly 😱

**Comment by kanongil** [comment]
That is one-off trial minutes. Also, those minutes would give us less then 42 CI-runs (with a 3 node x 3 platforms matrix).

**Comment by lloydbenson** [comment]
The TSC discussed this in our last meeting and we definitely are going to try out #2. @Nargonath is going to start a POC on one of the modules to see how it goes and I have started removing our reliance on travis from the hapi.dev site.

**Comment by Nargonath** [comment]
From what I found on GH website, there doesn't seem to be any limits for public repos in GH Action except for the limit of 20 concurrents jobs per repos.

> but workflows are a lot less maintainable

I think if we play it right by separating the different concerns between custom actions and leveraging the template feature we should have something generic enough. We'll see once I finished the PoC. We could also use a bot or web hooks to trigger build in other repo when we update the template if this is not a feature offered by the template feature.
